src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -496,6 +500,8 @@ class CLIJsonWebKeyCurveName(str, Enum):
         c.argument('identifier', options_list=['--id'], validator=validate_vault_or_hsm, help='Id of the HSM.')
         c.argument('resource_group_name', options_list=['--resource-group', '-g'],
                    help='Proceed only if HSM belongs to the specified resource group.')
+        c.argument('target_operation', arg_type=get_enum_type(CLISecurityDomainOperation),
```

Comment:
final command will be ?
```
az keyvault security-domain wait --target_operation upload
```

Reply:
`--target_operation` defaults to `upload` in avoid of breaking change. so either specifying `--target-operation` or not would work for `upload`.

for `download`, we have to specify `--target-operation download`